src/adapter/src/coord/command_handler.rs

```diff
@@ -1053,12 +1053,17 @@ impl Coordinator {
                 .controller
                 .storage
                 .monotonic_appender(entry.id())
+                .map_err(|_| name.clone())?;
+            let stats = coord
+                .controller
+                .storage
+                .webhook_statistics(entry.id())
```

Comment:
I did not know that `monotonic_appender` was only used by webhooks, I can definitely fold it in

Reply:
`MonotonicAppender` is a collection manager thing, and i don't want to pollute it with webhook stuff, so i'm gonna leave them separate

Reply:
That's fair. The issue I'm trying to point out more explicitly is that this is the only mention of `webhook` in the storage controller--this seems like we've gotten some part of the abstraction wrong here. What do you think is the right abstraction?

Reply:
Maybe another dimension of this is that if every caller of `monotonic_appender` also needs to call `webhook_statistics`, it seems preferable to find a means by which to bundle these things together.

Reply:
Not sure why this was marked as resolved without addressing the comment

Reply:
> The issue I'm trying to point out more explicitly is that this is the only mention of webhook in the storage controller--this seems like we've gotten some part of the abstraction wrong here. What do you think is the right abstraction?

Webhooks are fundamentally a special case here. They are "sources" that need to provide the storage controller with statistics from the adapter, as opposed to the communication channels with clusterd.

They happen to be the only users of `monotonic_appender`'s right now, but `monotonic_appender` is basically just a pass through channel to the collection manager controlled by the storage controller. We could lift the abstraction and declare monotonic appenders as intended specifically for special cases _like_ webhooks, where we do always want to report statistics, but thats not an abstraction decision I want to make as part of this pr, on a time crunch.

Reply:
What is the plan or the issue to revisit this to clean it up?

Reply:
I didn't make an explicit plan, because to me there is no urgency on resolving this tension; webhooks are likely going to be the only special case here, even in the future, and are eventually moving into clusterd

Reply:
I don't think this is great to just leave these strange bits dangling around without a coherent explanation in the code as to why these functions are special-cased/singular references/inconsistently named, nor to have a plan of ever aligning them. This doesn't make the code maintainable or grokkable IMO.

The urgency of resolving this tension is that no one is likely to fix this any time soon and we do not know how the code will drift around it. This means if anyone ever has a question about this inconsistency (which is highly likely), they will have to spelunk with `git annotate` to determine why things are in this shape. The best time to make an effort at resolving the tension––even with the most token effort––is before the code gets merged.